### PR TITLE
Use OS_VERSION when building rootfs image and update to noble for images

### LINKED ISSUE #6184: OS versions in versions.yaml are ignored for rootfs image
# Description of problem

OS_VERSION / assets.image.architecture.*.version is not used when building rootfs image

### PR DESCRIPTION
**Pull request by alex-matei**
Use OS_VERSION when building rootfs image. 

For Ubuntu builds "latest" was specified in versions.yaml and it was passed as IMG_OS_VERSION. That meant OS_VERSION was always empty and defaulted to "focal" in tools/osbuilder/rootfs-builder/ubuntu/config.sh. Changed "latest" to "jammy" to reflect the latest LTS version for Ubuntu.

Fixes #6184

### CODE REVIEW

**Review by amshinde — APPROVED** [review]
lgtm

File: versions.yaml [hunk 1]
```diff
@@ -121,13 +121,13 @@ assets:
     architecture:
       aarch64:
         name: "ubuntu"
-        version: "latest"
+        version: "focal"
```
Comment (liubin) [inline versions.yaml:124]: Should update to `jammy`, the latest LTS version, which equals `latest`.

Comment (alex-matei) [reply versions.yaml]: Updated to jammy.

**Comment by alex-matei** [comment]
I rebased on latest main branch and fixed merge conflicts. Please take a look.

**Comment by alex-matei** [comment]
@fidencio @gkurz Can you please take a look on this PR?

File: versions.yaml [hunk 2]
```diff
@@ -122,7 +122,7 @@ assets:
     architecture:
       aarch64:
         name: &default-image-name "ubuntu"
-        version: &default-image-version "latest"
+        version: &default-image-version "jammy"
```
Comment (gkurz) [inline versions.yaml:125]: Note that [latest LTS](https://fridge.ubuntu.com/2024/04/12/ubuntu-24-04-lts-noble-numbat-beta-released/) is expected later this week. It is worth to hold on a bit more and bump to `"noble"` when it is GA.

**Comment by gkurz** [comment]
Hi @alex-matei ! Please add a description in the second commit as required by CI.

**Pull request closed without merging**